Doc/library/ctypes.rst

```diff
@@ -2171,9 +2179,10 @@ Utility functions
 
 .. function:: POINTER(type, /)
 
-   Create and return a new ctypes pointer type. Pointer types are cached and
+   Create or return a ctypes pointer type. Pointer types are cached and
    reused internally, so calling this function repeatedly is cheap.
-   *type* must be a ctypes type.
+   *type* must be a ctypes-like type. The ctypes-like type is a type that
+   has class level attribute with the name ``__pointer_type__``.
```

Comment:
I think it's not preferable to mention that.

Reply:
- Let's not talk about “ctypes-like” types.
- I have no issue with mentioning where the cache is. However, I don't think you can cleanly use it.

```suggestion
   *type* must be a ctypes type.
   
   .. impl-detail::

      The resulting pointer type is cached in the ``__pointer_type__``
      attribute of *type*.
      It is possible to set this attribute before the first call to
      ``POINTER`` in order to set a custom pointer type.
      However, doing this is discouraged: manually creating a suitable
      pointer type is difficult without relying on implementation
      details that may change in future Python versions.
```